libraries/classes/ReplicationGui.php

```diff
@@ -356,20 +361,20 @@ public function getUsernameHostnameLength(): array
      *
      * @return string HTML code
      */
-    public function getHtmlForReplicationPrimaryAddReplicaUser(): string
+    public function getHtmlForReplicationPrimaryAddReplicaUser(?string $postUsername, ?string $hostname): string
     {
         [
             $usernameLength,
             $hostnameLength,
         ] = $this->getUsernameHostnameLength();
 
-        if (isset($_POST['username']) && strlen($_POST['username']) === 0) {
+        if ($postUsername !== null && strlen($postUsername) === 0) {
             $GLOBALS['pred_username'] = 'any';
         }
 
         $username = '';
-        if (! empty($_POST['username'])) {
-            $username = $GLOBALS['new_username'] ?? $_POST['username'];
+        if (! empty($postUsername)) {
```

Comment:
```suggestion
        if ($postUsername !== null && $postUsername !== '' && $postUsername !== '0') {
```
`empty` is bad because it hides typos. Please use more explicit check. In my experience the check for 0-string can be ignored, but please check the logic to make sure it's redundant.